tests/phpunit/tests/block-bindings/postMetaSource.php

```diff
@@ -0,0 +1,263 @@
+<?php
+/**
+ * Tests for Block Bindings API "core/post-meta" source.
+ *
+ * @package WordPress
+ * @subpackage Blocks
+ * @since 6.5.0
+ *
+ * @group blocks
+ * @group block-bindings
+ */
+class Tests_Block_Bindings_Post_Meta_Source extends WP_UnitTestCase {
+	protected static $post;
+	protected static $wp_meta_keys_saved;
+
+	/**
+	 * Modify the post content.
+	 *
+	 * @param string $content The new content.
+	 */
+	private function getModifiedPostContent( $content ) {
+		self::$post->post_content = $content;
+		// Update the global $post variable to ensure all tests get the correct $post context.
+		$this->updateGlobalPost();
+		return apply_filters( 'the_content', self::$post->post_content );
+	}
+
+	/**
+	 * Update the global $post variable.
+	 */
+	private function updateGlobalPost() {
```

Comment:
There isn't much value in having that code extracted from its method since it's used only once. It would make the code easier to follow if it was inlined.

You could also avoid adding inline comments by making the code structured like that:
```php
global $post;
$post = self::$post;
$post->post_content = $content;
return apply_filters( 'the_content', self::$post->post_content );
```

Reply:
The `self::$post` variable is useless anyway because the whole class only operates on the global post.

We can just always work on the global post if that is required.

Reply:
I've tried using the global $post variable only for all the tests, but something is not working as I expected.

It seems that the value of the global `$post` variable is `undefined` (or the value I defined in `wpSetUpBeforeClass`) for the FIRST tests. For the following tests, the global `$post` variable becomes always null unless I redefine it.

This happens even if I remove all the logic and I just access that variable.

The way I'm solving that is storing the $post object in a private property when the post is initialized: [link](https://github.com/WordPress/wordpress-develop/pull/6197/files#diff-b433ed8ede0ae232ea538e309b94dc6ff5d6debc4661ca993b791ad7f90b7c25R30). And for each test I assign the global variable to that: [link](https://github.com/WordPress/wordpress-develop/pull/6197/files#diff-b433ed8ede0ae232ea538e309b94dc6ff5d6debc4661ca993b791ad7f90b7c25R50).

However, this feels like a workaround. Any idea if that is expected or how it could be solved?